src/base/base_model.py

```diff
@@ -0,0 +1,47 @@
+from src.helpers.logger import LoggerFactory
+logger = LoggerFactory.get_logger(__name__)
+from src.config.settings import AppConfig
+from src.helpers.logger import LoggerFactory
+
+class BaseModel:
+    """
```

Comment:
This is a confusing name given `BaseModel` is commonly imported from Pydantic. I might suggest a name such as `PipelineStage` or similar.